client-sdk/ts-web/core/src/hdkey.ts

```diff
@@ -0,0 +1,234 @@
+import {eddsa} from 'elliptic';
```

Comment:
Wasn't the idea to use nacl?

Reply:
some of the hdwallet derivation steps needed direct access to some curve operations. we briefly talked about the idea of seeing if doing the same with tweetnacl internals would be possible. @Esya, have you looked into that?

Reply:
Elliptic only supports signing from 32-bytes seeds - called `k` or `private key` in [RFC 8032](https://tools.ietf.org/html/rfc8032) - but with BIP32-Ed25519, we have the post-hash private key (64 bytes)  - Elliptic is still needed for access to and manipulation of the 25519 Curve. 

I initially wanted to sign using `tweetnacl`, but it seems that the js implementation of NaCl does not allow to sign using the post-hash 64-bytes either, as their 64 bytes that they call *private keys* are actually 32-bytes seed/32-bytes public key to save time on computing the public key which is expensive. 

The paper mentions : 

> Due to the way child Ed25519 keys are derived, they will not have the 32-byte master secret key. We note that the NaCl library [3]provides the necessary interface to the extended private key.

[This article describes the differences between major libs](https://blog.mozilla.org/warner/2011/11/29/ed25519-keys/)

This [java implementation for instance](https://github.com/str4d/ed25519-java/blob/master/src/net/i2p/crypto/eddsa/spec/EdDSAPrivateKeySpec.java#L77) gets it right and allows instanciating directly from `h` (the post-hash 64-bytes signing key) 

But that does not seem to be the case for `tweetnacl` - I'd need further testing and comparison between the C implementation and the JS implementation, but I'm not a crypto expert and that'd be time consuming. It's unlikely but possible that this was overlooked when porting - so I've implemented the signing function myself, which I've tested verifies with all libs and with oasis-core. I'm fine reverting my previous PR if that makes more sense in the meantime.